src/cpp/shared/adventure/adv.cpp

```diff
@@ -122,11 +121,22 @@ void game::MakeAllWaterVisible(int player) {
 
 void advManager::DoEvent(class mapCell *cell, int locX, int locY) {
   hero *hro = &gpGame->heroes[gpCurPlayer->curHeroIdx];
-  int locType = cell->objType & 0x7F;
+  int LocationType = cell->objType & 0x7F;
   SAMPLE2 res2 = NULL_SAMPLE2;
-  ScriptCallback("OnLocationVisit", locType, locX, locY);
+  ScriptCallback("OnLocationVisit", LocationType, locX, locY);
+  this->HandleSpellShrine(cell, LocationType, hro, res2, locX, locY);
+  this->UpdateRadar(1, 0);
+  this->UpdateHeroLocators(1, 1);
+  this->UpdateTownLocators(1, 1);
+  this->UpdBottomView(1, 1, 1);
+  this->UpdateScreen(0, 0);
+  gpSoundManager->SwitchAmbientMusic(giTerrainToMusicTrack[this->currentTerrain]);
+  WaitEndSample(res2, res2.sample);
+  CheckEndGame(0, 0);
+}
 
-  switch (locType) {
+void advManager::HandleSpellShrine(class mapCell *cell, int LocationType, hero *hro, SAMPLE2 res2, int locX, int locY) {
```

Comment:
I made sure to move the logic for determining whether or not a location was a shrine or not (thus, determining whether or not to handle a spell shrine event) to the main modified DoEvent function, before the potential call of HandleSpellShrine. Sorry for the confusion